documents/oauth_token.md

```diff
@@ -0,0 +1,140 @@
+```
+
+-------------------------------------------------------------------------------------------------------
+### Get Access token
+```rb
+options = {
+    'client_id'          => '<YOUR_CLIENT_ID>',
+    'client_secret'     => '<YOUR_CLIENT_SECRET>',
+    'redirect_uri'      => 'https://example.com/razorpay_callback',
+    'grant_type'        => 'authorization_code',
+    'code'              => '<AUTHORIZATION_CODE>',
+    'mode'              => 'test'
+}
+oauth_token = Razorpay::OAuthToken.get_access_token(options)
+```
+
+**Parameters:**
+| Name           | Type   | Description                                                                                                                  |
+|----------------|--------|------------------------------------------------------------------------------------------------------------------------------|
+| client_id*     | string | Unique client identifier.                                                                                                    |
+| client_secret* | string | Client secret string.                                                                                                        |
+| redirect_uri*  | string | Specifies the same redirect_uri used in the authorisation request.                                                           |
+| grant_type*    | string | Defines the grant type for the request. Possible value are:<ul><li>authorization_code</li><li>client_credentials</li></ul>   |
+| code*          | string | Decoded authorisation code received in the last step. Note: Pass this parameter only when grant_type is 'authorization_code' |
+| mode           | string | The type of mode. Possible values: <ul><li>test</li><li>live (default)</li></ul>                                             |
+
+**Response:**
+```json
+{
+  "public_token": "rzp_test_oauth_9xu1rkZqoXlClS",
+  "token_type": "Bearer",
+  "expires_in": 7862400,
+  "access_token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImp0aSI6IkY1Z0NQYkhhRzRjcUpnIn0.eyJhdWQiOiJGNFNNeEgxanMxbkpPZiIsImp0aSI6IkY1Z0NQYkhhRzRjcUpnIiwiaWF0IjoxNTkyODMxMDExLCJuYmYiOjE1OTI4MzEwMTEsInN1YiI6IiIsImV4cCI6MTYwMDc3OTgxMSwidXNlcl9pZCI6IkYycVBpejJEdzRPRVFwIiwibWVyY2hhbnRfaWQiOiJGMnFQaVZ3N0lNV01GSyIsInNjb3BlcyI6WyJyZWFkX29ubHkiXX0.Wwqt5czhoWpVzP5_aoiymKXoGj-ydo-4A_X2jf_7rrSvk4pXdqzbA5BMrHxPdPbeFQWV6vsnsgbf99Q3g-W4kalHyH67LfAzc3qnJ-mkYDkFY93tkeG-MCco6GJW-Jm8xhaV9EPUak7z9J9jcdluu9rNXYMtd5qxD8auyRYhEgs",
+  "refresh_token": "def50200f42e07aded65a323f6c53181d802cc797b62cc5e78dd8038d6dff253e5877da9ad32f463a4da0ad895e3de298cbce40e162202170e763754122a6cb97910a1f58e2378ee3492dc295e1525009cccc45635308cce8575bdf373606c453ebb5eb2bec062ca197ac23810cf9d6cf31fbb9fcf5b7d4de9bf524c89a4aa90599b0151c9e4e2fa08acb6d2fe17f30a6cfecdfd671f090787e821f844e5d36f5eacb7dfb33d91e83b18216ad0ebeba2bef7721e10d436c3984daafd8654ed881c581d6be0bdc9ebfaee0dc5f9374d7184d60aae5aa85385690220690e21bc93209fb8a8cc25a6abf1108d8277f7c3d38217b47744d7",
```

Comment:
```suggestion
  "access_token": "eyJ0e...",
  "refresh_token": "def50...",
```

Reply:
The response was taken from razorpay public docs. This might easier for partners to understand